channel.go

```diff
@@ -758,6 +761,14 @@ func (ch *Channel) connectionCloseStateChange(c *Connection) {
 
 func (ch *Channel) onClosed() {
 	removeClosedChannel(ch)
+
+	select {
+	case <-ch.closed:
+		// Do nothing if already closed.
```

Comment:
Why would it already be closed? Only if we have a bug right? We can either:
 - always close (which would panic in prod if we closed an already closed channel). since channels aren't created dynamically anyway, this seems OK
 - log an error if the channel's already closed